### PR TITLE
fix: apply sanitizer to markdown preview

### PR DESCRIPTION
**Pull request by js87zz**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description
* applied sanitizer to markdown preview

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/editor/src/__test__/unit/sanitizer.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  it('removes unnecessary tags', () => {
+    expect(sanitizeHTML('<script>alert("test");</script>', true)).toBe('');
+    expect(sanitizeHTML('<embed>child alive</embed>', true)).toBe('child alive');
+    expect(sanitizeHTML('<object>child die</object>', true)).toBe('');
+    expect(sanitizeHTML('<details><summary>foo</summary></details>', true)).toBe('');
+    expect(sanitizeHTML('<input type="image" />', true)).toBe('');
+  });
+
+  describe('attributes', () => {
+    it('removes all attritube but white list', () => {
```
Comment (seonim-ryu) [inline apps/editor/src/__test__/unit/sanitizer.spec.ts:13]: `attritube` -> `attribute`로 수정해야겠네요ㅎ

File: apps/editor/src/sanitizer/htmlSanitizer.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ const reXSSAttrValue = /((java|vb|live)script|x):/i;
 const reOnEvent = /^on\S+/i;
 const reComment = /<!--[\s\S]*?-->/g;
 
-export function sanitizeHTML(html: string | Node, needHtmlText: boolean) {
+export function sanitizeHTML(html: string | Node, needHtmlText?: boolean) {
```
Comment (seonim-ryu) [inline apps/editor/src/sanitizer/htmlSanitizer.ts:49]: `needHtmlText`에 디폴트 값을 주는건 어떨까요??
```js
needHtmlText = false
```

File: apps/editor/src/sanitizer/htmlSanitizer.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ export function sanitizeHTML(html: string | Node, needHtmlText: boolean) {
   removeUnnecessaryTags(root);
   leaveOnlyWhitelistAttribute(root);
 
-  return domUtils.finalizeHtml(root, needHtmlText);
+  return domUtils.finalizeHtml(root, !!needHtmlText);
```
Comment (seonim-ryu) [inline apps/editor/src/sanitizer/htmlSanitizer.ts:62]: 위에 디폴트값 주면 형변환 해주지 않아도 되겠네요ㅎ

**Review by seonim-ryu — APPROVED** [review]
[2/2] 리뷰 완료합니다. 새니타이저도 간단하게 해결되었네요 👍

File: apps/editor/src/__test__/unit/sanitizer.spec.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        expect(sanitizeHTML('<a href="javascript:alert();">xss</a>', true)).toBe('<a>xss</a>');
+        expect(sanitizeHTML('<a href="  JaVaScRiPt: alert();">xss</a>', true)).toBe('<a>xss</a>');
+        expect(sanitizeHTML('<a href="vbscript:alert();">xss</a>', true)).toBe('<a>xss</a>');
+        expect(sanitizeHTML('<a href="  VBscript: alert(); ">xss</a>', true)).toBe('<a>xss</a>');
+        expect(sanitizeHTML('<a href="livescript:alert();">xss</a>', true)).toBe('<a>xss</a>');
+        expect(sanitizeHTML('<a href="  LIVEScript: alert() ;">xss</a>', true)).toBe('<a>xss</a>');
+        expect(sanitizeHTML(`123<a href=' javascript:alert();'>xss</a>`, true)).toBe(
+          '123<a>xss</a>'
+        );
+        expect(sanitizeHTML(`<a href='javas<!-- -->cript:alert()'>xss</a>`, true)).toBe(
```
Comment (lja1018) [inline apps/editor/src/__test__/unit/sanitizer.spec.ts:65]: 일부러 `s`와 `cript`를 분리하신건가요?? 아래는 합쳐있어서요!

**Review by lja1018 — APPROVED** [review]
리뷰완료
고생하셨습니다!

File: apps/editor/src/__test__/unit/sanitizer.spec.ts (hunk 4 above)
Comment (js87zz) [reply apps/editor/src/__test__/unit/sanitizer.spec.ts:65]: 넵~xss 테스트때문에 추가한 케이스입니다~!

**Comment by AdamGold** [comment]
When is this supposed to be released?

**Comment by js87zz** [comment]
@AdamGold 
This PR is for our next major release(`v3.0`).
It will be released within April.

**Pull request merged**